api/v1alpha1/user_types.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// UserSpec defines the desired state of S3User
+type UserSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Name of the S3User
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Password of the S3User
+	// +kubebuilder:validation:Optional
+	Password string `json:"password,omitempty"`
+
+	// Groups associated to the S3User
+	// +kubebuilder:validation:Optional
+	Groups []string `json:"groups,omitempty"`
```

Comment:
- Are Groups a "standard" S3 feature, or is it something specific to Minio ? In such a case, I'm not sure what would the proper way going forward could be : 
  - change User API so it can be generic ?
  - add a way (eg : flag ?) to toggle User reconciliation in main.go on/off depending on selected provider ?
  - reduce "s3-operator" scope to be "minio-operator" and abandon other potential future providers ?

- This applies to user groups and policies both : at the moment, these are managed as simple []string, but there are both S3 resources. Could we turn these []string into proper references to Policy API  (already existing) and Group API (which would then be TODO) ?

Opinions are welcome for all of my review points, but especially these two. :)

Reply:
i agree that management of group and policies should be symetric with a crd group probably, or abandon group

Reply:
I extracted the group part in a second PR